src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1706,3 +1705,13 @@ void C2_MacroAssembler::rvv_reduce_integral(Register dst, VectorRegister tmp,
 
   vmv_x_s(dst, tmp);
 }
+
+void C2_MacroAssembler::rvv_vsetvli(BasicType bt, int length_in_bytes, Register tmp) {
+  Assembler::SEW sew = Assembler::elemtype_to_sew(bt);
+  if (length_in_bytes != MaxVectorSize) {
+    mv(tmp, length_in_bytes / type2aelembytes(bt));
+    vsetvli(tmp, tmp, sew);
```

Comment:
We can make use of 'vsetivli' instruction when the number of elements to be processed is not bigger than 31. That will help save one 'mv' instruction in this case.

Reply:
> We can make use of 'vsetivli' instruction when the number of elements to be processed is not bigger than 31. That will help save one 'mv' instruction in this case.

Thanks, it has been fixed.